src/safeposix/syscalls/net_calls.rs

```diff
@@ -1597,50 +1921,81 @@ impl Cage {
             newbuflen -= bytecount;
             newbufptr = newbufptr.wrapping_add(bytecount);
 
             //if we're not still peeking data, consume the data we peeked from our peek
             // buffer and if the bytecount is more than the length of the peeked
             // data, then we remove the entire buffer
+            // ** What is the point of the if else statement being passed in as the
+            // parameter ?? bytecount = min(sockhandle.last_peek.len(),
+            // newbuflen) so how can bytecount be greater than
+            // sockhandle.last_peek.len() ?? Seems like it always returns
+            // bytecount regardless ** //
             if flags & MSG_PEEK == 0 {
                 let len = sockhandle.last_peek.len();
                 sockhandle
                     .last_peek
                     .drain(..(if bytecount > len { len } else { bytecount }));
             }
 
+            //if we've filled all of the buffer with peeked data, return with success
             if newbuflen == 0 {
-                //if we've filled all of the buffer with peeked data, return
-                return bytecount as i32;
+                return bytecount as i32; //return number of bytes read into
+                                         // buff
             }
         }
 
+        //Initialize variables to indicate a pointer to the first available
+        //byte in the buff and remaining buffer length, respectively
         let bufleft = newbufptr;
         let buflenleft = newbuflen;
         let mut retval;
 
+        //The domain of the socket is UNIX
+        //lind handles UNIX communication using pipes
         if sockhandle.domain == AF_UNIX {
             // get the remote socket pipe, read from it, and return bytes read
+            //
+            //Check if the socket is non-blocking
+            //If no messages are available at the socket, the receive calls
+            //wait for a message to arrive, unless the socket is nonblocking
+            //(see fcntl(2)), in which case the value -1 is returned and errno
+            //is set to EAGAIN or EWOULDBLOCK.
             let mut nonblocking = false;
             if sockfdobj.flags & O_NONBLOCK != 0 {
                 nonblocking = true;
             }
+            //we loop here so we can cancel blocking recvs, if necessary
             loop {
+                //Grab the receive pipe from the socket to read the data
+                //into the remaining space in the buffer **
                 let sockinfo = &sockhandle.unix_info.as_ref().unwrap();
                 let receivepipe = sockinfo.receivepipe.as_ref().unwrap();
                 retval = receivepipe.read_from_pipe(bufleft, buflenleft, nonblocking) as i32;
+                //In the case of an error from reading from the receive pipe
                 if retval < 0 {
+                    //** This step seems weird
+                    // Even if the call fails we return with the previous peeked data? */
```

Comment:
The idea of recv is that as long as it already reads something, it can return. The read length does not need to be the same as the passed buffer length, and that's the meaning of the recv return value. But if we did not read anything, we must block until something gets read